Galaxus.Functional/(Either)/Either2.cs

```diff
@@ -131,6 +132,42 @@ public T Match<T>(Func<A, T> onA, Func<B, T> onB)
             }
         }
 
+        /// <summary>
+        /// An overload for <see cref="Match"/> using async functions.
+        /// </summary>
+        /// <param name="onA">Async function to be called when field "A" is in use. The argument to this action is never the <b>null</b> reference.</param>
+        /// <param name="onB">Async function to be called when field "B" is in use. The argument to this action is never the <b>null</b> reference.</param>
+        public Task<T> MatchAsync<T>(Func<A, Task<T>> onA, Func<B, Task<T>> onB)
+        {
+            return Match(
+                async a => await onA(a),
+                async b => await onB(b));
```

Comment:
Wouldn't it be better here (and in all other instances) to rather just return the task instead of awaiting it in order to avoid the overhead from the async/await state machine?

E.g.:
```csharp
return Match(onA, onB);
```

Reply:
It's true that you gain a little bit of performance by directly returning the task, however awaiting makes diagnostics and exception handling easier and is good practice in general (see also [Prefer async / await over directly returning Task](https://github.com/davidfowl/AspNetCoreDiagnosticScenarios/blob/master/AsyncGuidance.md#prefer-asyncawait-over-directly-returning-task)). In the context of IO, which async should ultimately be used for, the overhead of the async / await state machine should be negligible compared to the overhead of IO.

Reply:
Interesting, I did not know that the stack trace was different in these two approaches. But would it then not make more sense to await the return value of the `Match(...)` method instead of the anonymous lambda functions? That seems to result in a better stack trace.

Reply:
I agree 👍 Done (also for Result.MatchAsync)